storage/resource_mount.go

```diff
@@ -37,6 +39,34 @@ func (cb mountCallback) preProcess(r *schema.Resource) func(
 	}
 }
 
+// If a cluster_id is specified for the mount, this function validates that the
+// cluster actually exists. If the cluster does not exist, we assume (incorrectly)
+// that the mount also does not exist.
+//
+// In cases of update requests this would lead to an attempted recreation of the
+// mount
+// In cases of delete, this could orphan the mount
+//
+// This is needed to resolve issue: https://github.com/databricks/terraform-provider-databricks/issues/1864
+// because without a running cluster we do not have a way to do CRUD on mounts.
+// The argument being that possibly orphaning mounts is a better experience than
+// requiring manual edits to tfstate.
```

Comment:
What would happen if we only set the `cluster_id` to the empty string if it doesn't exist?

Is it possible that might trigger creation of a temporary cluster to perform the CRUD after all?

Reply:
Do you need an instance profile to read mounts? If so, I think this is a good option. We only need the temporary cluster for the read pathway.

Reply:
Re: are instance profiles required? 
Not sure. One disadvantage that this approach has over the current approach is though users will need cluster_creation permissions to apply. This could be a regression for some customers.

Re: Setting cluster_id to the empty string. 
I believe that is what happens in the current code path. If the cluster_id is empty then the provider tries to use an instance profile if you have one defined. 
see: 
https://github.com/databricks/terraform-provider-databricks/blob/8dc13d4d703fee2ad85b6c60fedba72f0b0b87b1/storage/s3.go#L70
AND
https://github.com/databricks/terraform-provider-databricks/blob/8dc13d4d703fee2ad85b6c60fedba72f0b0b87b1/storage/s3.go#L79